src/data/scripts/campaign/MPC_coronaResistScript.kt

```diff
@@ -32,20 +32,64 @@ open class MPC_coronaResistScript(
         return false
     }
 
+    private fun iterateThroughAffected(){
+        val iter = affecting.iterator()
+        val fleetBin: MutableSet<CampaignFleetAPI> = HashSet()
+
+        while (iter.hasNext()){
+            val it = iter.next()
+            var shouldRemove=false
+            if((it.fleetData==null)||(it.fleetData.fleet==null)){
+                shouldRemove=true
+            }
+            else if (!shouldAffectFleet((it.fleetData.fleet))){
+                if (!fleetBin.contains(it.fleetData.fleet)){
+                    fleetBin.add(it.fleetData.fleet)
```

Comment:
also i can do this myself, but kotlin standard substitutes list.add for list += entry

```suggestion
                    fleetBin += it.fleetData.fleet
```

Reply:
(or at least, i do)